app/models/user.rb

```diff
@@ -45,6 +45,26 @@ def set_address
     self.address = "#{city}, #{country}" if city.present? && country.present?
   end
 
+  def address_format
+    if address.blank?
+      errors.add(:address, "can't be blank")
+    else
+      parts = address.split(',')
+      if parts.length != 2 || parts.any?(&:blank?)
+        errors.add(:address, 'must include a city and country separated by a comma')
+      end
+    end
+  end
+
+  def validate_address_is_real_place
+    return unless address.present?
+
+    geocode = Geocoder.search(address).first
+    errors.add(:address, 'must correspond to a real place') unless geocode && geocode.coordinates.present?
+  rescue StandardError => e
+    errors.add(:address, 'could not be validated')
+  end
```

Comment:
Same idea here, address shouldn't be modifiable. The error handing is good, we can add it in the set_address method
```
  def set_address
    self.address = "#{city}, #{country}" if city.present? && country.present?
    errors.add(:city, 'must correspond to a real city') unless Geocoder.search(address).first
  end
```